Bottom: 2ef7290b84f2f295677916d95e93b4ee64305673
Top:    3018d0714f61907190e023eef99e42077c036f78
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-25 10:17:59 -0500

Small fixes to improve idempotency

Generally, adding changed_when parameters or tweaking tasks to ensure changed
statuses are reported correctly and tasks only run when needed.


---
diff --git a/roles/bacula-storage/tasks/main.yml b/roles/bacula-storage/tasks/main.yml
index 01125af..1a6db49 100644
--- a/roles/bacula-storage/tasks/main.yml
+++ b/roles/bacula-storage/tasks/main.yml
@@ -18,11 +18,21 @@
     mode: 0640
     backup: yes
 
+# - name: Check if baculatape selinux module exists
+#   stat: 
+#     path: /etc/selinux/targeted/modules/active/modules/baculatape.pp
+#   register: baculatape_pp
+# - import_tasks: selinux-tape-policy.yml
+#   when: baculatape_pp.stat.exists is defined and baculatape_pp.stat.exists == false
+
 - name: Check if baculatape selinux module exists
-  stat: path=/etc/selinux/targeted/modules/active/modules/baculatape.pp
+  command: semodule -l 
   register: baculatape_pp
+  changed_when: false
+
 - import_tasks: selinux-tape-policy.yml
-  when: baculatape_pp.stat.exists is defined and baculatape_pp.stat.exists == false
+  when: "'baculatape\t1.0' not in baculatape_pp.stdout_lines"
+
 - name: Make sure bacula user is in the tape group
   user: 
     name: bacula
@@ -30,6 +40,8 @@
     append: yes
 - name: Test Bacula Storage configuration
   command: /usr/sbin/bacula-sd -t -c {{ bacula_storage_config_dir }}/bacula-sd.conf
+  changed_when: false
+
 - name: Enable and start Bacula Storage
   service: 
     name: bacula-sd
